src/main/java/seedu/address/model/affiliation/AffiliationChecker.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.model.affiliation;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.affiliation.exceptions.AffiliationPersonNotFoundException;
+import seedu.address.model.affiliation.exceptions.SamePersonAffiliationException;
+import seedu.address.model.affiliation.exceptions.SameRoleAffiliationException;
+import seedu.address.model.person.Person;
+
+/**
+ * AffiliationChecker checks if affiliation is valid.
+ */
+public class AffiliationChecker {
+
+    /**
+     * Returns true if affiliation list pass the check, throws an exception otherwise.
+     *
+     * @param model The current model that is running.
+     * @throws CommandException If affiliation list fails the check.
+     */
+    public static Boolean check(Person personAddingAffiliation, Model model) throws CommandException {
```

Comment:
Maybe you can rename this to `authenticateAffiliation`?

Reply:
Sure will make the changes in the next commit.